qiskit/providers/fake_provider/generic_backend_v2.py

```diff
@@ -406,7 +411,12 @@ def _build_generic_target(self):
                 noise_params = self._get_noise_defaults(name, gate.num_qubits)
                 self._add_noisy_instruction_to_target(gate, noise_params, calibration_inst_map)
             else:
-                self._target.add_instruction(gate, properties=None, name=name)
+                if is_fully_connected:
+                    self._target.add_instruction(gate, properties=None, name=name)
+                else:
+                    qarg_set = self._coupling_map if gate.num_qubits > 1 else range(self.num_qubits)
+                    props = {(qarg,) if isinstance(qarg, int) else qarg: None for qarg in qarg_set}
+                    self._target.add_instruction(gate, properties=props, name=name)
```

Comment:
hmmm what makes the fully connected cmap special in this case? Is this purely an efficiency matter?

Reply:
Yep, it is much more efficient to set `properties=None' here in that case, otherwise we create a dictionary with entries for each edge of the fully connected coupling map that all point to None...

Reply:
I see, but I am not sure I like this approach. This would make the behavior differ "arbitrarily" depending on the connectivity of the coupling map used, right? Some times the instructions would be global, some times they would be local with no properties... I think that the behavior should be consistent independently of the connectivity.

Reply:
I see - the issue here is that the memory overhead is quite large when representing an all-to-all connected by local instructions without properties. It can be up to ~2 GB for large backends whereas a backend with global instructions and no properties needs a couple of MB. Granted, one should likely not create a large all-to-all connected backend but if a user does, he will be surprised by that memory overhead. 

However, I agree that a discrete change based on that fully-connected flag is similarly confusing - a backend missing only one edge to be all-to-all connected would exhibit a different memory behaviour than a fully-connected backend. Should I remove the optimizations based on `is_fully_connected`? I think the pre-PR behaviour of noiseless `BackendV2` is definitely incorrect...

Reply:
I think that it's safer to remove the optimizations for `is_fully_connected`. It's the default behavior of the class and I can see how users would get confused if specifying a custom cmap suddenly affects the gate properties.